src/Apps/Sell/Routes/ArtistRoute.tsx

```diff
@@ -0,0 +1,178 @@
+  const submission = useFragment(FRAGMENT, props.submission)
+
+  return <ArtistRoute submission={submission} />
+}
+
+export const ArtistRoute: React.FC<{
+  submission?: ArtistRoute_submission$data
+}> = ({ submission }) => {
+  const { router } = useRouter()
+  const { actions } = useSellFlowContext()
+
+  const isNewSubmission = !submission?.internalID
+
+  const onSubmit = async () => {}
+
+  const createSubmission = async (artist: AutocompleteArtist) => {
+    if (!artist?.internalID) return
+
+    const isTargetSupply = artist?.targetSupply?.isTargetSupply
+
+    if (!isTargetSupply) {
+      router.push(`/sell2/artist-not-eligible/${artist.internalID}`)
+      return
+    }
+
+    const response = await actions.createSubmission({
+      artistID: artist.internalID,
+    })
+    const submissionID =
+      response?.createConsignmentSubmission?.consignmentSubmission?.externalId
+
+    router.replace(`/sell2/submissions/${submissionID}/artist`)
+    router.push(`/sell2/submissions/${submissionID}/title`)
+  }
+
+  const updateSubmission = async (artist: AutocompleteArtist) => {
+    if (!artist?.internalID) return
+
+    const isTargetSupply = artist?.targetSupply?.isTargetSupply
+    if (!isTargetSupply) {
+      return
+    }
+
+    await actions.updateSubmission({ artistID: artist.internalID })
+    actions.goToNextStep()
+  }
+
+  const onSelect = async (
+    artist: AutocompleteArtist,
+    formik: FormikProps<FormValues>
+  ) => {
+    if (!artist?.internalID) return
+
+    const isTargetSupply = artist?.targetSupply?.isTargetSupply
+    formik.setFieldValue("isTargetSupply", isTargetSupply)
+
+    if (isNewSubmission) {
+      createSubmission(artist)
+    } else {
+      updateSubmission(artist)
+    }
+  }
+
+  const initialValues: FormValues = {
+    artistName: submission?.artist?.name || "",
+    artistId: submission?.artist?.internalID || "",
+    isTargetSupply: submission?.artist?.targetSupply?.isTargetSupply || false,
+  }
+
+  return (
+    <Formik<FormValues>
+      initialValues={initialValues}
+      onSubmit={onSubmit}
+      validateOnMount
+      validationSchema={Schema}
+    >
+      {formik => (
+        <SubmissionLayout hideNavigation={isNewSubmission}>
+          <Text mb={2} variant="xl">
+            Add artist name
+          </Text>
+
+          <ArtistAutoComplete
+            onSelect={artist => {
+              onSelect(artist, formik)
+            }}
+            onError={() => console.error("something happened")}
+            showChevronIcon
+            title="Artist"
+          />
+
+          <Text mt={2} variant="sm" color="black60">
+            Currently, artists can not sell their own work on Artsy.{" "}
+            <RouterLink to="https://support.artsy.net/s/article/Im-an-artist-Can-I-submit-my-own-work-to-sell">
+              Learn more.
+            </RouterLink>
+          </Text>
+
+          {!isNewSubmission &&
```

Comment:
If the user edits the artist of an already existing submission, we show the message inline instead of sending the user to the "Artist not eligible" page so they can continue editing the submission.